src/main/java/henry/Parser.java

```diff
@@ -0,0 +1,121 @@
+package henry;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import command.Command;
+import command.DeadlineCommand;
+import command.DeleteCommand;
+import command.EchoCommand;
+import command.EventCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.TodoCommand;
+import command.UnmarkCommand;
+import exceptions.HenryException;
+
+/**
+ * The parser class is responsible for parsing the user input and
+ * creating the appropriate command.
+ */
+public class Parser {
+
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<command>\\S*)(?<args>.*)");
+    private static final Pattern DEADLINE_FORMAT = Pattern.compile("(?<desc>.+) /by "
+                                                                   + "(?<dateTime>\\d{2}-\\d{2}-\\d{4} "
+                                                                   + "\\d{2}:\\d{2})");
+
+    private static final Pattern EVENT_FORMAT = Pattern.compile("(?<desc>.+) /at "
+                                                                + "(?<dateTime>\\d{2}-\\d{2}-\\d{4} "
+                                                                + "\\d{2}:\\d{2})");
+    private static final String DATE_FORMATTER = "dd-MM-yyyy HH:mm";
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMATTER);
+
+    /**
+     * Parses a command from a string. There are two types of commands:
+     * single-argument commands, and double-argument commands. The function
+     * first checks if the command matches the appropriate pattern. If it does,
+     * then the corresponding command is returned. If not, then an exception is thrown.
+     *
+     * @param text the string to be parsed.
+     * @return the command corresponding to the string, or an exception if the string is malformed.
+     */
+    public Command parseCommand(String text) {
+        Matcher matcher = BASIC_COMMAND_FORMAT.matcher(text.trim());
+        if (!matcher.matches()) {
+            throw new HenryException("UNKNOWN COMMAND!");
+        }
+        String command = matcher.group("command").trim();
+        String args = matcher.group("args").trim();
+        switch (command) {
+        case EchoCommand.COMMAND_WORD:
+            return new EchoCommand(args);
+        case MarkCommand.COMMAND_WORD:
+            if (isArgsFormattedCorrectly(args)) {
+                throw new HenryException("ARGUMENT IS NOT A NUMBER!");
+            }
+            return new MarkCommand(Integer.parseInt(args));
+        case UnmarkCommand.COMMAND_WORD:
+            if (isArgsFormattedCorrectly(args)) {
```

Comment:
Iirc, try to write everything in full? So Args should be Arguments

Reply:
Modified in latest.